src/status_im2/contexts/chat/bottom_sheet_composer/gesture.cljs

```diff
@@ -0,0 +1,103 @@
+(ns status-im2.contexts.chat.bottom-sheet-composer.gesture
+  (:require
+    [react-native.gesture :as gesture]
+    [react-native.reanimated :as reanimated]
+    [oops.core :as oops]
+    [status-im2.contexts.chat.bottom-sheet-composer.constants :as c]
+    [status-im2.contexts.chat.bottom-sheet-composer.utils :as utils]
+    [utils.re-frame :as rf]))
+
+(defn set-opacity
+  [e opacity translation expanding? min-height max-height new-height saved-height]
+  (let [remaining-height     (if @expanding?
+                               (- max-height (reanimated/get-shared-value saved-height))
+                               (- (reanimated/get-shared-value saved-height) min-height))
+        progress             (if (= new-height min-height) 1 (/ translation remaining-height))
+        currently-expanding? (neg? (oops/oget e "velocityY"))
+        max-opacity?         (and currently-expanding? (= (reanimated/get-shared-value opacity) 1))
+        min-opacity?         (and (not currently-expanding?)
+                                  (= (reanimated/get-shared-value opacity) 0))]
+    (if (>= translation 0)
+      (when (and (not @expanding?) (not min-opacity?))
+        (reanimated/set-shared-value opacity (- 1 progress)))
+      (when (and @expanding? (not max-opacity?))
+        (reanimated/set-shared-value opacity (Math/abs progress))))))
+
+(defn maximize
+  [{:keys [maximized?]}
+   {:keys [height saved-height background-y opacity]}
+   {:keys [max-height]}]
+  (reanimated/animate height max-height)
+  (reanimated/set-shared-value saved-height max-height)
+  (reanimated/set-shared-value background-y 0)
+  (reanimated/animate opacity 1)
+  (reset! maximized? true)
+  (rf/dispatch [:chat.ui/set-input-maximized true]))
+
+(defn minimize
+  [{:keys [input-ref emoji-kb-extra-height saved-emoji-kb-extra-height]}]
+  (when @emoji-kb-extra-height
+    (reset! saved-emoji-kb-extra-height @emoji-kb-extra-height)
+    (reset! emoji-kb-extra-height nil))
+  (rf/dispatch [:chat.ui/set-input-maximized false])
+  (.blur ^js @input-ref))
+
+(defn bounce-back
+  [{:keys [height saved-height opacity background-y]}
+   {:keys [window-height]}
+   starting-opacity]
+  (reanimated/animate height (reanimated/get-shared-value saved-height))
+  (when (zero? starting-opacity)
+    (reanimated/animate opacity 0)
+    (reanimated/animate-delay background-y (- window-height) 300)))
+
+(defn drag-gesture
+  [{:keys [input-ref] :as props}
+   {:keys [gesture-enabled?] :as state}
+   {:keys [height saved-height last-height opacity background-y container-opacity] :as animations}
+   {:keys [max-height lines] :as dimensions}
+   keyboard-shown]
+  (let [expanding?       (atom true)
+        starting-opacity (reanimated/get-shared-value opacity)]
+    (->
```

Comment:
Another small comment about style: this `->` should have its first parameter right after:

```clojure
(-> a ;; here
    b
    c)
```

Reply:
Can you elaborate please?

Reply:
I meant, you have:

```clojure
(->
   a 
   b
   c
```

instead of:

```clojure
(-> a 
    b
    c
```
```